sdk/core/azure-core/azure/core/credentials_async.py

```diff
@@ -46,3 +50,37 @@ async def __aexit__(
         traceback: Optional[TracebackType] = None,
     ) -> None:
         pass
+
+
+@runtime_checkable
+class AsyncSupportsTokenInfo(Protocol, AsyncContextManager["AsyncSupportsTokenInfo"]):
+    """Protocol for classes able to provide OAuth access tokens with additional properties."""
+
+    async def get_token_info(self, *scopes: str, options: Optional[_TokenRequestOptions] = None) -> _AccessTokenInfo:
```

Comment:
Because credentials_async is public.

We don't want user to use 
from azure.core.credentials_async import AccessToken

Reply:
Pyright/Pylance considers imported symbols [private by default](https://github.com/microsoft/pyright/blob/main/docs/typed-libraries.md#library-in). So trying to import `AccessToken` would cause it to complain: `"AccessToken" is not exported from module "azure.core.credentials_async"
  Import from "azure.core.credentials" instead`. Would you still prefer to keep the underscore here regardless?

Reply:
Yes. We cannot assume users all have pyright/pylance enabled. :)